addons/web_editor/static/lib/odoo-editor/src/OdooEditor.js

```diff
@@ -2819,16 +2821,16 @@ export class OdooEditor extends EventTarget {
         // If the element have a class,
         // we only remove the attribute to ensure we don't break some style.
         // Otherwise we remove the entire inline element.
-        for (const emptyElement of element.querySelectorAll('[oe-zws-empty-inline]')) {
-            if (emptyElement.textContent.length === 1 && emptyElement.textContent.includes('\u200B')) {
-                if (emptyElement.classList.length > 0) {
-                    emptyElement.removeAttribute('oe-zws-empty-inline');
+        for (const el of element.querySelectorAll('[oe-zws-empty-inline]')) {
```

Comment:
I would agree with `cleanZWS`.
But if avoid creating new functions is a must, I can just transform the snippet below into a one-liner and use it twice:
```
 [el, ...descendants(el)]
            .filter(node => node.nodeType === Node.TEXT_NODE)
            .forEach(node => node.nodeValue = node.nodeValue.replace(/\u200B/g, ''));
```       
(Actually it already is, it was just spread in 3 lines for readability).
But it could be rewritten as 
`[element, ...descendants(element)].forEach(node => node.nodeValue && node.nodeValue.replace(/\u200B/g, ''));`
I consider it to be a bit less readable, but it it's idiomatic in our code base and does save lines.

About your suggestion about merging the two querySelectorAll in one:
FA-elements would satisfy the first `if` condition (`isZWS`) and never reach that outer `else` where the ZWS cleaning happen. So I'd rather keep them separated (it's also more readable, right?).

Reply:
Avoiding creating new functions is not a must per se. If we agree on the name `cleanZWS` then I think it is a valuable util function and we should therefore keep it. 👍